base/strings/basic.jl

```diff
@@ -48,6 +48,8 @@ sizeof(s::AbstractString) = error("type $(typeof(s)) has no canonical binary rep
 eltype{T<:AbstractString}(::Type{T}) = Char
 
 (*)(s1::AbstractString, ss::AbstractString...) = string(s1, ss...)
+(.*){V,S<:AbstractString}(v::Vector{V},s::S) = [i*s for i in v]
```

Comment:
Now `V` can be of any type, not just strings.  I think you want:
```jl
(.*){S<:AbstractString}(v::Vector{S},s::AbstractString) = [i*s for i in v]
```
Note that you don't need a type parameter for the `s` argument.